provider/resource_keycloak_openid_client_service_account_role.go

```diff
@@ -40,18 +40,32 @@ func resourceKeycloakOpenidClientServiceAccountRole() *schema.Resource {
 	}
 }
 
-func getOpenidClientServiceAccountRoleFromData(data *schema.ResourceData) *keycloak.OpenidClientServiceAccountRole {
-	return &keycloak.OpenidClientServiceAccountRole{
-		Id:                   data.Id(),
-		ContainerId:          data.Get("client_id").(string),
-		Name:                 data.Get("role").(string),
-		RealmId:              data.Get("realm_id").(string),
-		ServiceAccountUserId: data.Get("service_account_user_id").(string),
+func getOpenidClientServiceAccountRoleFromData(data *schema.ResourceData, keycloakClient *keycloak.KeycloakClient) (*keycloak.OpenidClientServiceAccountRole, error) {
+
+	containerId := data.Get("client_id").(string)
+	roleName := data.Get("role").(string)
+	realmId := data.Get("realm_id").(string)
+	serviceAccountRoleId := data.Get("service_account_user_id").(string)
+
+	if realmId != "" && containerId != "" {
+		role, err := keycloakClient.GetRoleByName(realmId, containerId, roleName)
+		if err != nil {
+			return nil, err
+		}
+		return &keycloak.OpenidClientServiceAccountRole{
+			Id:                   role.Id,
+			ContainerId:          containerId,
+			Name:                 roleName,
+			RealmId:              realmId,
+			ServiceAccountUserId: serviceAccountRoleId,
+		}, nil
 	}
+
+	return nil, nil
 }
 
 func setOpenidClientServiceAccountRoleData(data *schema.ResourceData, serviceAccountRole *keycloak.OpenidClientServiceAccountRole) {
-	data.SetId(serviceAccountRole.Id)
+	data.SetId(fmt.Sprintf("%s/%s", serviceAccountRole.ServiceAccountUserId, serviceAccountRole.Id))
```

Comment:
This might be a breaking change since the ID for this resource is being changed.  Can you verify that these resources aren't being recreated because of this change?

Reply:
Just new resources will be created. In context of service account roles it'll try to apply existing changes once to add service account roles with ned IDs but POST action to service account roles endpoint always returns 204 even if role was already added